src/log.rs

```diff
@@ -1,38 +1,94 @@
-//! Code for setting up the program logger.
-use env_logger::Env;
+/// The `log` module provides initialisation and configuration of the application’s logging system.
+///
+/// This module sets up logging with various levels (error, warn, info, debug, trace) and optional
+/// colourisation based on terminal support. It also allows configuration of the log level through
+/// environment variables.
+///
+use atty;
+use chrono::Local;
+use fern::colors::{Color, ColoredLevelConfig};
+use fern::Dispatch;
```

Comment:
These can be amalgamated:

```suggestion
use fern::colors::{Color, ColoredLevelConfig, Dispatch};
```